### ISSUE TITLE
VPS on Amazon EC2

### ISSUE THREAD

**Issue by Wilt8086** [Ysurac/openmptcprouter-vps#36]
Hi,

Trying to setup VPS on Amazon EC2, the instance always are unreachable after succesful install (Debian 10 stock) and reboot. As far i can see from console and screen captures when booting with the new mptcp kernel, seems the instance hangs on booting kernel. Maybe it needs some modules enabled like the debian-cloud kernel version.

There is any other kernel version I can use to make it work on EC2? Anyone has succesfully booted VPS on EC2? Also i look for kernel openmptcprouter sources to compile a custom kernel version, but haven't found it.

I know EC2 isn't on the list of cloud services tested, but will be a great thing if we can make it work on Amazon.

Thanks!

**Comment by Ysurac** [comment]
This was already reported but I have no idea what is exactly needed to make it work on EC2.
I will update VPS kernel compile doc, there is that: https://github.com/Ysurac/openmptcprouter-vps/wiki/Compile-kernel but really too old.

**Comment by Wilt8086** [comment]
Hi Ysurac,

I've been able to boot the VPS with your current mptcp kernel on EC2, but only on not ENA enabled instances: C4, M4, D2 or T2. ENA is the enhanced network adapter EC2 uses on new instance types.

So seems the only required thing will be add the ENA module to the kernel, as the other important difference seems ok: new instances use KVM virtualization, and the old ones (C4, M4, D2 and T2) use XEN, but both seems well supponted on openmptcprouter kernel.

The Amazon ENA drivers are here https://github.com/amzn/amzn-drivers to compile as a module. If i can  try to compile a current version of openmptcprouter kernel with the ENA driver, and let see if it works.

Regards,

**Comment by Ysurac** [comment]
ok, thanks for the info :) 
I will update the doc today and I will try to compile these drivers.

**Comment by Wilt8086** [comment]
Oh, that will be great! Let me know how it goes.

Thanks!

**Comment by Wilt8086** [comment]
Hi Ysurac,

Tested on EC2 on a C5 instance with ENA enabled, latest VPS 0.1021 on Debian 10. Works great!
Installed flawlessly and after rebooting, working really good.

Thanks for including the ENA driver on the kernel, was the missing piece :)

**Comment by Ysurac** [comment]
Really ? I didn't had success with Ubuntu on a t2 instance...

**Comment by Wilt8086** [comment]
Yes, for sure. Debian 10 works, not tested with Ubuntu yet. Maybe you can try a t3 instance, these have ENA enabled and KVM virtualization. T2 are on XEN and no ENA. After install, only opened ports on security group and running.

Also tested on C4 instance (XEN and no ENA) and also works fine under Debian 10.

**Comment by Wilt8086** [comment]
Just tested with Ubuntu and you're right. Don't boot. Keeps rebooting just after start loading kernel, and loops to grub boot screen forever. Something with grub config on ubuntu? At boot grub has selected and try to boot the menu option "Advanced options for Ubuntu" but i don't know if it's normal when an external kernel is added.

**Comment by hairy-dog** [comment]
What sort of money does Amazon EC2 cost for this? I'm baffled by their pricing: a rough ball-park estimate of what, say 100GB through in a month would cost.
